google/cloud/bigtable/mutations.py

```diff
@@ -13,41 +13,200 @@
+            raise ValueError(
+                "timestamp_micros must be positive (or -1 for server-side timestamp)"
+            )
+        self.family = family
+        self.qualifier = qualifier
+        self.new_value = new_value
+        self.timestamp_micros = timestamp_micros
+
+    def _to_dict(self) -> dict[str, Any]:
+        """Convert the mutation to a dictionary representation"""
+        return {
+            "set_cell": {
+                "family_name": self.family,
+                "column_qualifier": self.qualifier,
+                "timestamp_micros": self.timestamp_micros,
+                "value": self.new_value,
+            }
+        }
+
+    def is_idempotent(self) -> bool:
+        """Check if the mutation is idempotent"""
+        return self.timestamp_micros != SERVER_SIDE_TIMESTAMP
 
 
 @dataclass
 class DeleteRangeFromColumn(Mutation):
     family: str
-    column_qualifier: bytes
-    start_timestamp_ms: int
-    end_timestamp_ms: int
+    qualifier: bytes
+    # None represents 0
+    start_timestamp_micros: int | None = None
+    # None represents infinity
+    end_timestamp_micros: int | None = None
+
+    def __post_init__(self):
+        if (
+            self.start_timestamp_micros is not None
+            and self.end_timestamp_micros is not None
+            and self.start_timestamp_micros > self.end_timestamp_micros
+        ):
+            raise ValueError("start_timestamp_micros must be <= end_timestamp_micros")
+
+    def _to_dict(self) -> dict[str, Any]:
+        timestamp_range = {}
+        if self.start_timestamp_micros is not None:
+            timestamp_range["start_timestamp_micros"] = self.start_timestamp_micros
+        if self.end_timestamp_micros is not None:
+            timestamp_range["end_timestamp_micros"] = self.end_timestamp_micros
+        return {
+            "delete_from_column": {
+                "family_name": self.family,
+                "column_qualifier": self.qualifier,
+                "time_range": timestamp_range,
+            }
+        }
 
 
 @dataclass
 class DeleteAllFromFamily(Mutation):
     family_to_delete: str
 
+    def _to_dict(self) -> dict[str, Any]:
+        return {
+            "delete_from_family": {
+                "family_name": self.family_to_delete,
+            }
+        }
+
 
 @dataclass
 class DeleteAllFromRow(Mutation):
-    pass
+    def _to_dict(self) -> dict[str, Any]:
+        return {
+            "delete_from_row": {},
+        }
 
 
-@dataclass
-class BulkMutationsEntry:
-    row: bytes
-    mutations: list[Mutation] | Mutation
+class RowMutationEntry:
+    def __init__(self, row_key: bytes | str, mutations: Mutation | list[Mutation]):
+        if isinstance(row_key, str):
+            row_key = row_key.encode("utf-8")
+        if isinstance(mutations, Mutation):
+            mutations = [mutations]
+        self.row_key = row_key
+        self.mutations = tuple(mutations)
+
+    def _to_dict(self) -> dict[str, Any]:
```

Comment:
maybe in a future PR it might be good to make the mutations list mutable so that end users can build up the mutation array

Reply:
I opened an issue to track this here: https://github.com/googleapis/python-bigtable/issues/795